### PR TITLE
bugfix rate input validation and fix the unit tests

### PR DESCRIPTION
**Pull request by alramdein**
- There is a bug in input validation for `rate()` function in `Attacker` class. Previously the code was: 
  ```
  rate(value) {
          if (!isValidRateStr(value) || !isInteger(value) || value < 0) {
              throw TypeError('rate must be a positive integer or a valid flexRate string (ex: 50/m)');
          }
          return this.addFlag('rate', value);
      }
  ```
  This will make every input are false. If the input is string, it'll not satisfy the `isInteger()` and if the input is integer, it'll not satisfy the `isValidRateStr()`

- The unit test seems to be not properly finished. The export for `encoder` in file `encoder.test.js` is false due to the capital in the first letter. And the `should return instance of plot` calling the method `encode()`

- I also added the `body.json` inside the `/tests` folder to support the testing so it runs well. Also I added a `package.lock.json` since it is important and the `.gitignore` not ignoring it anyways, strange.

I planned to use this library, so I hope you can response it as soon as possible. Thank you!

### CODE REVIEW

**Comment by jbarabander** [comment]
Hey there!  Thanks for taking an interest in this library and for fixing some of the unfortunately glaringly obvious problems.  I haven't maintained in awhile but seeing as there has been some interest in it I will try to put some time into maintaining it more.

File: src/utils.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ let validateResolver = (resolver) => {
 let rateStrRegex = /\d+\/(ns|us|µs|ms|s|m|h)/;
 
 let isValidRateStr = (rate) => {
-    if (typeof rate !== 'string') {
+    if (typeof rate !== 'string' && !(rate instanceof String)) {
```
Comment (jbarabander) [inline src/utils.js:35]: NIT: I don't think this check is totally necessary since it is generally considered bad practice to use the String class.

**Comment by jbarabander** [comment]
As for the fix itself I only have one small comment but its fairly nitpicky so I suppose I will merge this.

**Comment by jbarabander** [comment]
Merged! Nice work man!  I will publish a new version of it on npm soon!

File: src/utils.js (hunk 1 above)
Comment (alramdein) [reply src/utils.js:35]: I referenced to [this](https://stackoverflow.com/questions/4059147/check-if-a-variable-is-a-string-in-javascript#:~:text=I%20strenuously%20disagree,at%204%3A11) StackOverflow discussion. Which is I think it is better to handle all possibility. But yeah you right that using String class is actually a _very unusual_ coding practice.

**Comment by alramdein** [comment]
Thank you very much for the fast response! Keep to great work! Looking forward for the new version release.

File: src/utils.js (hunk 1 above)
Comment (jbarabander) [reply src/utils.js:35]: Ahh got it!  Works for me and yeah it makes sense.

**Pull request merged**